components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -811,19 +811,27 @@ def _get_prepared_image(request, iid, server_id=None, conn=None,
     img = conn.getObject("Image", iid)
     if img is None:
         return
-    reverses = _get_maps_enabled(r, 'reverse', img.getSizeC())
+    invert_flags = None
+    if 'maps' in r:
+        reverses = _get_maps_enabled(r, 'reverse', img.getSizeC())
+        # 'reverse' is now deprecated (5.4.0). Also check for 'invert'
+        invert_flags = _get_maps_enabled(r, 'inverted', img.getSizeC())
+        # invert is True if 'invert' OR 'reverse' is enabled
+        if reverses is not None and invert_flags is not None:
```

Comment:
Should this be `and invert_flags is None` to match the logic of `setActiveChannels`?

Reply:
No, I need to check if both are None here since ```zip(invert_flags , reverses)``` will fail if either is ```None```. If there's no ```"reverse"``` in the request `maps` query string then ```reverses = [None, None]``` here. The following line ```z[0] if z[0] is not None else z[1]``` is where I'm checking the "invert_flags" first, then using "reverse" if invert is not used.

Reply:
Thank you for the explanation and I'm certainly splitting hairs at this point, but I'm not sure what would need to happen for this to occur. I'd think if `invert_flags` is not None, then no further processing is needed.

Reply:
In the case where url is ```...&maps=[{"inverted":{"enabled":true}},{"reverse":{"enabled":true}}]``` then we want to take into account both the 'inverted' and 'reverse' maps.
```
invert_flags = [True, None]
reverses = [None, True]
```
So, we cannot ignore ```reverses``` even if ```invert_flags is not None```.

Or in this case, where we are using the old url:
```...&maps=[{"reverse":{"enabled":true}},{"reverse":{"enabled":true}}]```
then
```
invert_flags = [None, None]
reverses = [True, True]
```
So even now, ```invert_flags is not None```, but we are zipping and comparing with reverses for each channel in turn.

Reply:
The combination of inverted/reverse works.
 tested yesterday
Maybe a robot test for the combination?